src/client/content_mapblock.cpp

```diff
@@ -513,10 +513,10 @@ f32 MapblockMeshGenerator::getCornerLevel(int i, int k)
 			count++;
 		} else if (content == CONTENT_AIR) {
 			air_count++;
-			if (air_count >= 2)
-				return -0.5 * BS + 0.2;
 		}
 	}
+	if (air_count >= 2)
```

Comment:
Why move this out of the loop? An early return is preferrable for performance.

Reply:
I think this is to improve the liquid detection and is intended as-is. The loop will now return `0.5 * BS` instead of the `air` case if there was a liquid or source detected.